internal/api/connectors/inventory/xjoin.go

```diff
@@ -0,0 +1,129 @@
+package inventory
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/Khan/genqlient/graphql"
+	"github.com/spf13/viper"
+)
+
+type XjoinConnectorClient interface {
+	GetHostDetailsRequest(
+		ctx context.Context,
+		IDs []string,
+		hostsFilter HostFilter,
+		orderBy HOSTS_ORDER_BY,
+		orderHow ORDER_DIR,
+		limit int,
+		offset int,
+	) (response []*HostDetailsResponse, err error)
+
+	GetSystemProfileDetailsRequest(
+		ctx context.Context,
+		IDs []string,
+		hostsFilter HostFilter,
+		orderBy HOSTS_ORDER_BY,
+		orderHow ORDER_DIR,
+		limit int,
+		offset int,
+	) (response []*SystemProfileDetailsResponse, err error)
+}
+
+type XjoinConnectorClientImpl struct {
+	client XjoinConnectorClient //nolint unused
+}
+
+func createHostFilterList(IDs []string) []*HostFilter {
+	hostFilterList := make([]*HostFilter, len(IDs))
+	for i, ID := range IDs {
+		hostFilterList[i] = &HostFilter{Id: FilterStringWithWildcard{Eq: ID}}
+	}
+
+	return hostFilterList
+}
+
+func createHostDetailResponseList(resp *getHostDetailsResponse) []*HostDetailsResponse {
+	hostDetailResponseList := make([]*HostDetailsResponse, len(resp.Hosts.Data))
+	for i, response := range resp.Hosts.Data {
+		hostDetailResponseList[i] = &HostDetailsResponse{
+			ID:             response.Id,
+			Facts:          response.Facts,
+			CanonicalFacts: response.Canonical_facts,
+		}
+	}
+
+	return hostDetailResponseList
+}
+
+func createSystemProfileResponseList(resp *getSystemProfileFactsResponse) []*SystemProfileDetailsResponse {
+	systemProfileResponseList := make([]*SystemProfileDetailsResponse, len(resp.Hosts.Data))
+	for i, response := range resp.Hosts.Data {
+		systemProfileResponseList[i] = &SystemProfileDetailsResponse{
+			ID:                 response.Id,
+			SystemProfileFacts: response.System_profile_facts,
+		}
+	}
+
+	return systemProfileResponseList
+}
+
+func createHostsFilter(IDs []string) HostFilter {
+	return HostFilter{
+		OR: createHostFilterList(IDs),
+	}
+}
+
+func NewXjoinClientWithRequestDoer(cfg *viper.Viper) graphql.Client {
```

Comment:
I don't think this is necessary.  Although, we need to provide a "factory" method that creates an instance of the `XjoinConnectorClientImpl` struct.